src/backend/renderer/utils/wayland.rs

```diff
@@ -69,11 +113,10 @@ impl RendererSurfaceState {
                 self.buffer_scale = attrs.buffer_scale;
                 self.buffer_transform = attrs.buffer_transform.into();
 
-                if let Some(old_buffer) = std::mem::replace(&mut self.buffer, Some(buffer)) {
-                    if &old_buffer != self.buffer.as_ref().unwrap() {
-                        old_buffer.release();
-                    }
+                if !self.buffer.as_ref().map(|b| b == buffer).unwrap_or(false) {
```

Comment:
I think this could be more concisely (and readably IMHO) written:
`if !self.buffer.as_ref().map_or(false, |b| b == &buffer) {`

(maybe someday we get is_some_and)